src/EnumServiceProvider.php

```diff
@@ -34,13 +36,21 @@ public function register(): void
         ]);
 
         $this->registerRequestTransformMacro();
+        $this->registerRouterBindingMacros();
     }
 
     protected function registerRequestTransformMacro(): void
     {
         Request::mixin(new EnumRequest);
     }
 
+    protected function registerRouterBindingMacros(): void
+    {
+        Router::macro('enumSubstituteImplicitBindings', function ($route) {
+            ImplicitRouteBinding::resolveForRoute($this->container, $route);
```

Comment:
I'm not sure if we need to add a dedicated class to do this and add a macro to `Router`.

I did this 'cause it passes `$this->container`, which is inside `Routers`' s context. I'm not sure if we can just inline it inside the middleware since service container should be a singleton within whole application?